src/test/java/org/jvnet/hudson/plugins/platformlabeler/ConfigurationTest.java

```diff
@@ -0,0 +1,102 @@
+    nodeLabelCache = ComputerListener.all().get(NodeLabelCache.class);
+  }
+
+  @Test
+  public void configuredOnlyOneLabel() {
+    PlatformLabelerNodeProperty nodeProperty = new PlatformLabelerNodeProperty();
+    LabelConfig labelConfig = new LabelConfig();
+    labelConfig.setArchitecture(false);
+    labelConfig.setArchitectureName(false);
+    labelConfig.setArchitectureNameVersion(false);
+    labelConfig.setVersion(false);
+    labelConfig.setNameVersion(false);
+    nodeProperty.setLabelConfig(labelConfig);
+    r.jenkins.getNodeProperties().add(nodeProperty);
+
+    nodeLabelCache.onConfigurationChange();
+    Set<LabelAtom> labelsAfter = computer.getNode().getAssignedLabels();
+    System.err.println(labelsAfter);
+    assertThat(labelsAfter.size(), is(2));
+  }
+
+  @Test
+  public void configuredTwoLabels() {
+    PlatformLabelerNodeProperty nodeProperty = new PlatformLabelerNodeProperty();
+    LabelConfig labelConfig = new LabelConfig();
+    labelConfig.setArchitectureName(false);
+    labelConfig.setArchitectureNameVersion(false);
+    labelConfig.setName(false);
+    labelConfig.setNameVersion(false);
+    nodeProperty.setLabelConfig(labelConfig);
+    r.jenkins.getNodeProperties().add(nodeProperty);
+
+    nodeLabelCache.onConfigurationChange();
+    Set<LabelAtom> labelsAfter = computer.getNode().getAssignedLabels();
+    System.err.println(labelsAfter);
+    assertThat(nodeLabelCache.getLabelsForNode(computer.getNode()).size(), is(2));
+    assertThat(labelsAfter.size(), is(3));
+  }
+
+  @Test
+  public void configuredAllLabels() {
+    PlatformLabelerNodeProperty nodeProperty = new PlatformLabelerNodeProperty();
+    LabelConfig labelConfig = new LabelConfig();
+    nodeProperty.setLabelConfig(labelConfig);
+    r.jenkins.getNodeProperties().add(nodeProperty);
+
+    nodeLabelCache.onConfigurationChange();
+    Set<LabelAtom> labelsAfter = computer.getNode().getAssignedLabels();
+    System.err.println(labelsAfter);
+    assertThat(labelsAfter.size(), is(7));
+  }
+
+  @Test
+  public void nodeConfigOverridesGlobalConfig() {
+
+    PlatformLabelerGlobalConfiguration globalConfig =
+        GlobalConfiguration.all().getInstance(PlatformLabelerGlobalConfiguration.class);
+
+    LabelConfig globalLabelConfig = new LabelConfig();
+
+    globalLabelConfig.setVersion(false);
+    globalLabelConfig.setArchitecture(false);
+
+    globalConfig.setLabelConfig(globalLabelConfig);
+
+    PlatformLabelerNodeProperty nodeProperty = new PlatformLabelerNodeProperty();
+    LabelConfig labelConfig = new LabelConfig();
+    labelConfig.setVersion(false);
+    nodeProperty.setLabelConfig(labelConfig);
+    r.jenkins.getNodeProperties().add(nodeProperty);
+
+    nodeLabelCache.onConfigurationChange();
+    Set<LabelAtom> labelsAfter = computer.getNode().getAssignedLabels();
+    System.err.println(labelsAfter);
+    assertThat(nodeLabelCache.getLabelsForNode(computer.getNode()).size(), is(5));
+    assertThat(labelsAfter.size(), is(6));
```

Comment:
Could this also assert the contents of the set rather than the size of the set?

Reply:
done